toke.c

```diff
@@ -791,6 +791,21 @@ S_scan_terminated(pTHX_ char *s, I32 ival) {
     return s;
 }
 
+#define POP_OLDBUFPTR TRUE
+
+STATIC void
+S_no_op_when_xoperator(pTHX_ const char *const what, char *s, int pop_oldbufptr) {
+    if (PL_expect != XOPERATOR)
+        return;
+
+    if (pop_oldbufptr && PL_bufptr > s) {
+        s = PL_bufptr - 1;
+        PL_bufptr = PL_oldbufptr;
+    }
+
```

Comment:
This could just be moved to S_no_op(), it is static, so not an API, so we can change its signature.

All but one caller to no_op() checks for `XOPERATOR` immediately before the call, and the one exception simply checks a bit earlier.

So I think this check could just be folded into no_op().

Reply:
It's probably worth renaming no_op() if this is done, I'm not especially fond of no_op_when_xoperator(), maybe check_op_expected(), but other suggestions are welcome.

Reply:
re: move to `S_no_op` ... done 10dedbbdfb
re: name ... I'm not very good with names :-) what about `warn_expect_operator` ?